pkg/kubelet/cm/devicemanager/plugin/v1beta1/stub.go

```diff
@@ -144,13 +149,33 @@ func (m *Stub) Start() error {
 	return nil
 }
 
+func (m *Stub) Restart() error {
+	klog.InfoS("Restarting Device Plugin server")
+	if m.server == nil {
+		return nil
+	}
+	// Send terminate signal to ListAndWatch()
+	m.termSignal <- true
+
+	m.server.Stop()
+	m.server = nil
+
+	m.Start()
+	return m.Start()
+}
+
 // Stop stops the gRPC server. Can be called without a prior Start
 // and more than once. Not safe to be called concurrently by different
 // goroutines!
 func (m *Stub) Stop() error {
+	klog.InfoS("Stopping device plugin server")
 	if m.server == nil {
 		return nil
 	}
+	// Send terminate signal to ListAndWatch()
+	m.termSignal <- true
+	m.stopWatcher <- true
```

Comment:
can we extract helpers from `Start` and `Stop` and reimplement `Restart` composing these helpers or there are complications which prevent or make this approach unpractical?

Reply:
Ack, I will play around with that, we could potentially prevent some repetition of code by using helpers.

Reply:
I tried creating a helper that it can be called in both `Restart` and `Stop` but that caused unit tests failures so decided to keep the code as is for now. I think the unit tests are relying on certain behavior of `Stop` and `Start` methods so changing them is causing some issues. If you feel strongly about this, I can do to deep dive into the specific unit tests and invest more time into this as it is not as straightforward as I thought.